tfe/data_source_agent_pool.go

```diff
@@ -36,7 +36,7 @@ func dataSourceTFEAgentPoolRead(d *schema.ResourceData, meta interface{}) error
 	options := tfe.AgentPoolListOptions{}
 
 	for {
-		l, err := tfeClient.AgentPools.List(ctx, organization, options)
+		l, err := tfeClient.AgentPools.List(ctx, organization, &options)
```

Comment:
Why not pass `nil` in here and other places where ListOptions are not assigning any particular values?

Reply:
nvm! I just noticed that further in the code we need to assign a value to `options.PageNumber`